### PR TITLE
Fix roundPixel bug with bitmap texts, issue #6934

### PR DESCRIPTION
**Pull request by shaise**
##### Description of change
Fix roundPixel bug with bitmap texts, issue #6934

##### Pre-Merge Checklist

- [x] Tests and/or benchmarks are included
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

[TestRoundPixel.zip](https://github.com/pixijs/pixi.js/files/5441697/TestRoundPixel.zip)

### CODE REVIEW

File: packages/text-bitmap/src/BitmapText.ts [hunk 1]
```diff
@@ -474,7 +474,14 @@ export class BitmapText extends Container
...
             const char = chars[i];
-            const xPos = (char.position.x + lineAlignOffsets[char.line]) * scale;
+            let offset = char.position.x + lineAlignOffsets[char.line];
+
+            if (this.roundPixels)
+            {
+                offset = Math.floor(offset);
+            }
+
+            const xPos = offset * scale;
```
Comment (SukantPal) [inline packages/text-bitmap/src/BitmapText.ts:484]: What if `scale` is a non-integer?

Comment (shaise) [reply packages/text-bitmap/src/BitmapText.ts:484]: When scaling / rotating a font you loose aliasing anyway. The roundPixels is good only if you work with 1.0 scale / 0.0 rotation.

File: packages/text-bitmap/src/BitmapText.ts [hunk 2]
```diff
@@ -474,7 +474,14 @@ export class BitmapText extends Container
         for (let i = 0; i < lenChars; i++)
         {
             const char = chars[i];
-            const xPos = (char.position.x + lineAlignOffsets[char.line]) * scale;
+            let offset = char.position.x + lineAlignOffsets[char.line];
+
+            if (this.roundPixels)
+            {
+                offset = Math.floor(offset);
```
Comment (bigtimebuddy) [inline packages/text-bitmap/src/BitmapText.ts:481]: Any reason for `floor` over `round`?

Comment (shaise) [reply packages/text-bitmap/src/BitmapText.ts]: You are right. round is better. I have modified the code

File: packages/text-bitmap/src/BitmapText.ts (hunk 1 above)
Comment (bigtimebuddy) [reply packages/text-bitmap/src/BitmapText.ts:484]: This sounds reasonable to me.

**Review by bigtimebuddy — APPROVED** [review]

**Review by SukantPal — APPROVED** [review]
Thanks @shaise!

**Comment by bigtimebuddy** [comment]
@themoonrat could you take a quick look at this?

**Review by themoonrat — APPROVED** [review]

**Comment by themoonrat** [comment]
> @themoonrat could you take a quick look at this?

I'm happy with this. Before commit cfc05d1e91d198f7d1fdddc6ff62f39e5965e353 - bitmap text contained individual for each character, each of which had the sprite's 'roundPixels' property set.

But in that commit, BitmapText got changed to using a Mesh for optimisation reasons, which lost any usage of 'roundPixels' altogether. I'm happy that this PR effectively adds that back in.

**Comment by themoonrat** [comment]
What I'll add, tho, is that the roundPixels is no longer dynamic, like sprites are, where you could turn it on and off dynamically and see it's effects at runtime. Now, it's dependent on roundPixels being set before new text is set. 

Not worth this PR being held up, imo but potential for a follow up pr for public getter and setter on roundPixels to then also call updateText function here.

**Comment by bigtimebuddy** [comment]
That's a good suggestion to use setter/getter for this. Would be nice to have round work after text is set. @shaise is this something you want to add? No worries if you don't, we can add it later.

**Comment by shaise** [comment]
Hi @bigtimebuddy ,

Shouldn't be an issue to add a getter/setter. I will do it and update the PR. 

shai.

**Comment by shaise** [comment]
getter/setter added. attached demo showing the fix in action.
[test_round_pixels.zip](https://github.com/pixijs/pixi.js/files/5451305/test_round_pixels.zip)

**Review by themoonrat — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
text
(example above toggling roundPixels)

Looks good @shaise, thanks for that fix :+1:

**Pull request merged**